samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -89,7 +98,7 @@ public String getChangelogStream(String storeName) {
     if (systemStreamRes != null) {
       systemStreamRes = StreamManager.createUniqueNameForBatch(systemStreamRes, this);
```

Comment:
Not related to your change. But `getChangelogStream` is not a pure value-accessor contrary to its name. Would be better to move the creation outside of this get method.

Reply:
Are you concerned about the call to a method called "create"? `createUniqueNameForBatch` just looks up some configs; it doesn't actually have a side effect.